packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -1050,21 +1063,148 @@ class RenderParagraph extends RenderBox
             final TextRange originalRange = infoAttribute.range;
             attributes.add(
               infoAttribute.copy(
-                  range: TextRange(start: offset + originalRange.start,
-                      end: offset + originalRange.end)
+                range: TextRange(
+                  start: offset + originalRange.start,
+                  end: offset + originalRange.end,
+                ),
               ),
             );
           }
           buffer.write(label);
           offset += label.length;
         }
-        _cachedAttributedLabel = AttributedString(buffer.toString(), attributes: attributes);
+        _cachedAttributedLabels = <AttributedString>[AttributedString(buffer.toString(), attributes: attributes)];
       }
-      config.attributedLabel = _cachedAttributedLabel!;
+      config.attributedLabel = _cachedAttributedLabels![0];
       config.textDirection = textDirection;
     }
   }
 
+  ChildSemanticsConfigurationsResult _childSemanticsConfigurationsDelegate(List<SemanticsConfiguration> childConfigs) {
+    if (_cachedAttributedLabels == null) {
+      final ChildSemanticsConfigurationsResult result = _buildChildSemanticsConfigurationsResultAndFillCache(childConfigs);
+      assert(() {
+        // Verify the cache will produce the same result.
+        final ChildSemanticsConfigurationsResult resultBuiltFromCache = _buildChildSemanticsConfigurationsResultAndFillCache(childConfigs);
+        assert(result.siblingMergeGroups.isEmpty && resultBuiltFromCache.siblingMergeGroups.isEmpty);
+        assert(result.mergeUp.length == resultBuiltFromCache.mergeUp.length);
+        int childConfigsIndex = 0;
+        for (int index = 0; index < result.mergeUp.length; index += 1) {
+          if (identical(result.mergeUp[index], resultBuiltFromCache.mergeUp[index])) {
+            // This must be coming from the input `childConfigs`
+            assert(identical(result.mergeUp[index], childConfigs[childConfigsIndex]));
+            childConfigsIndex += 1;
+          } else {
+            assert(result.mergeUp[index].label == resultBuiltFromCache.mergeUp[index].label);
+          }
+        }
+        // All the ``childConfigs` must be included in the merge up.
+        assert(childConfigs.length == childConfigsIndex);
+        return true;
+      }());
+      return _buildChildSemanticsConfigurationsResultAndFillCache(childConfigs);
```

Comment:
Should this just return result instead of calling _buildChildSemanticsConfigurationsResultAndFillCache again?